src/code/z_player_lib.c

```diff
@@ -2348,8 +3237,597 @@ void Player_DrawGreatFairysMask(PlayState* play, Player* player) {
+                    static Vec3f D_801C0D6C = { 420.0f, 1210.0f, 380.0f };
+                    Vec3s* temp_s1;
+                    Vec3f* var_a0 = &D_801C0D60;
+
+                    if (player->transformation == PLAYER_FORM_HUMAN) {
+                        var_a0 = &D_801C0D6C;
+                    }
+
+                    Matrix_MultVec3f(var_a0, &heldActor->world.pos);
+                    Matrix_RotateZYX(0x69E8, -0x5708, 0x458E, MTXMODE_APPLY);
+
+                    Matrix_Get(&sp230);
+                    temp_s1 = &heldActor->world.rot;
+                    Matrix_MtxFToYXZRot(&sp230, temp_s1, false);
+                    heldActor->shape.rot = *temp_s1;
+                } else if (player->stateFlags1 & PLAYER_STATE1_800) {
+                    heldActor->world.rot.y = heldActor->shape.rot.y =
+                        player->actor.shape.rot.y + player->leftHandWorld.rot.y;
+                }
+            } else {
+                static f32 sMeleeWeaponLengths[PLAYER_MELEEWEAPON_MAX] = {
+                    0.0f,    // PLAYER_MELEEWEAPON_NONE
+                    3000.0f, // PLAYER_MELEEWEAPON_SWORD_KOKIRI
+                    3000.0f, // PLAYER_MELEEWEAPON_SWORD_RAZOR
+                    4000.0f, // PLAYER_MELEEWEAPON_SWORD_GILDED
+                    5500.0f, // PLAYER_MELEEWEAPON_SWORD_GREAT_FAIRY
+                    -1.0f,   // PLAYER_MELEEWEAPON_STICK
+                    2500.0f, // PLAYER_MELEEWEAPON_ZORA_FINS
+                };
+
+                if ((player->transformation == PLAYER_FORM_FIERCE_DEITY) ||
+                    ((player->transformation != PLAYER_FORM_ZORA) &&
+                     ((player->itemAction == PLAYER_IA_STICK) ||
+                      ((player->meleeWeaponState != 0) &&
+                       (player->meleeWeaponAnimation != PLAYER_MWA_GORON_PUNCH_RIGHT) &&
+                       (player->meleeWeaponAnimation != PLAYER_MWA_GORON_PUNCH_BUTT))))) {
+                    if (player->itemAction == PLAYER_IA_STICK) {
+                        D_801C0994->x = player->unk_B08[1] * 5000.0f;
+                    } else {
+                        D_801C0994->x = sMeleeWeaponLengths[Player_GetMeleeWeaponHeld(player)];
+                    }
+                    func_80126B8C(play, player);
+                }
+
+                Matrix_Get(&player->mf_CC4);
+                Matrix_MtxFToYXZRot(&player->mf_CC4, &player->leftHandWorld.rot, false);
+            }
+        }
+    } else if (limbIndex == PLAYER_LIMB_RIGHT_HAND) {
+        Actor* sp224 = player->heldActor;
+        s32 pad;
 
-#pragma GLOBAL_ASM("asm/non_matchings/code/z_player_lib/func_80128BD0.s")
+        if (*dList1 != NULL) {
+            if (player->rightHandType == PLAYER_MODELTYPE_RH_BOW) {
+                static Gfx* D_801C0D94 = object_link_child_DL_017818;
+                static Vec3f D_801C0D98 = { -35.0f, -395.0f, 0.0f };
+                // Unused
+                static f32 D_801C0DA4 = 0.0f;
```

Comment:
This is the last data of the file, so could you just get rid of this? Should just be file padding

Reply:
Almost, there's `sRightHandLimbModelShieldQuadVertices` after this one

Reply:
Why a float?